xbmc/interfaces/builtins/PlayerBuiltins.cpp

```diff
@@ -39,6 +41,34 @@
 #include "Autorun.h"
 #endif
 
+namespace
+{
+bool IsMediasourceRestricted(const CFileItem item)
+{
+  const std::shared_ptr<CProfileManager> profileManager =
+      CServiceBroker::GetSettingsComponent()->GetProfileManager();
```

Comment:
GetSettingsComponent returns a pointer, so theres a possibility of it being a nullptr.
I know this style is used everywhere, but i think we should be more defensive in general (eg check getsettignscomponent, log nullptr/fail method). Just my thoughts.

Reply:
ok, i'm thinking of
```c
if (!componentSettings)
{
 log(LOGERROR, "peng.. nullptr {}" , __func__);
 return false; // claim this mediasource is not locked, just catch it in thje log
}
something better you have in mind?
```

Reply:
Essentially, yeah that'd be fine